src/manager.cpp

```diff
@@ -237,11 +266,178 @@ static const std::map<pid_t, std::string> start_modules(const std::vector<Module
         // we can only come here, if we're the parent!
         auto child_pid = handle.check_child_executed();
 
-        EVLOG_verbose << fmt::format("Forked module {} with pid: {}", module.name, child_pid);
-        children[child_pid] = module.name;
+        EVLOG_debug << fmt::format("Forked module {} with pid: {}", module.name, child_pid);
+        started_modules[child_pid] = module.name;
+    }
+
+    return started_modules;
+}
+
+struct ModuleReadyInfo {
+    bool ready;
+    std::shared_ptr<TypedHandler> token;
+};
+
+// FIXME (aw): these are globals here, because they are used in the ready callback handlers
+std::map<std::string, ModuleReadyInfo> modules_ready;
+std::mutex modules_ready_mutex;
+
+static std::map<pid_t, std::string> start_modules(Everest::Config& config, Everest::MQTTAbstraction& mqtt_abstraction,
```

Comment:
This should be ::Everest::Config and ::Everest::MQTTAbstraction otherwise it won't compile with clang with the following error:
error: reference to 'Everest' is ambiguous

Reply:
ah, good to know